src/wp-admin/includes/class-wp-media-list-table.php

```diff
@@ -503,15 +503,22 @@ public function column_title( $post ) {
 	 * Handles the author column output.
 	 *
 	 * @since 4.3.0
+	 * @since 6.8.0 Added check for empty author and fallback display
 	 *
 	 * @param WP_Post $post The current WP_Post object.
 	 */
 	public function column_author( $post ) {
-		printf(
-			'<a href="%s">%s</a>',
-			esc_url( add_query_arg( array( 'author' => get_the_author_meta( 'ID' ) ), 'upload.php' ) ),
-			get_the_author()
-		);
+		$author = get_the_author();
+
+		if ( ! empty( $author ) ) {
+			printf(
+				'<a href="%s">%s</a>',
+				esc_url( add_query_arg( array( 'author' => get_the_author_meta( 'ID' ) ), 'upload.php' ) ),
+				esc_html( $author )
+			);
+		} else {
+			echo '<span aria-hidden="true">&#8212;</span><span class="screen-reader-text">' . esc_html__( '(no author)', 'default' ) . '</span>';
```

Comment:
```suggestion
			echo '<span aria-hidden="true">&#8212;</span><span class="screen-reader-text">' . __( '(no author)' ) . '</span>';
```

Core translations are [considered safe](https://core.trac.wordpress.org/ticket/58900#comment:2), and 'default' for the `$domain` is unnecessary.